### PR TITLE
Fixes #34363 - Improve error reporting

### PR DESCRIPTION
**Pull request by adamruzicka**
Includes:
- https://github.com/theforeman/smart_proxy_remote_execution_ssh/pull/71

Transitively requires:
- https://github.com/theforeman/smart_proxy_dynflow/pull/103

### CODE REVIEW

**Comment by adamruzicka** [comment]
The polling script runner relies on several tools being available on the remote machine (curl, gnu coreutils), maybe it would be a good idea to check their presence in preflight checks? :thinking:

File: lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb [hunk 1]
```diff
@@ -126,7 +125,11 @@ def load_event_updates(event_data)
     end
 
     def cleanup
-      run_sync("rm -rf \"#{remote_command_dir}\"") if @cleanup_working_dirs
+      if @cleanup_working_dirs
+        assert_remote_command("rm -rf \"#{remote_command_dir}\"",
```
Comment (ezr-ondrej) [inline lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb:129]: what is `assert_remote_command`? assert should not run anything in my mind... it should just checks if something had run (had happend)...

File: lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb [hunk 2]
```diff
@@ -380,9 +382,16 @@ def upload_file(local_path, remote_path)
...
-      exit_code, _output, err = run_sync("mkdir -p #{path}")
-      if exit_code != 0
-        raise "Unable to create directory on remote system #{path}: exit code: #{exit_code}\n #{err}"
+      assert_remote_command("mkdir -p #{path}",
+        publish: true,
+        error: "Unable to create directory #{path} on remote system, exit code: %{exit_code}"
+      )
+    end
+
+    def assert_remote_command(cmd, error: nil, **kwargs)
```
Comment (ezr-ondrej) [inline lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb:391]: Can we rename it to `ensure...` ?

**Review by ezr-ondrej — COMMENTED** [review]
Thoughts: :exploding_head: 

I'll probably need to go from the dynflow PR to wrap my head around it, but if we could come up with little pieces of documentation while at it, it would be awesome (personally I believe we have quite a lot of code and we could get rid of some so I'm sad we are adding some, but it's just a feeling :)) )

File: lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb:129]: Right, it is "run the thing, check if it succeeded and raise an exception if not". If you thing `ensure_*` would depict it better then I'll change it

Comment (ezr-ondrej) [reply lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb:129]: I think so, it is not great, but IMO bit better then assure

Comment (ezr-ondrej) [inline lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb:129]: This is still assert, I'd expect it would fail, right?

Comment (adamruzicka) [reply lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb:129]: Correct

**Comment by ezr-ondrej** [comment]
ok, lets resume here ^_^

**Comment by adamruzicka** [comment]
rebased

File: lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb (hunk 1 above)
Comment (adamruzicka) [reply lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb]: Oh, I misunderstood the comment previously, sorry about this

File: lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb [hunk 3]
```diff
@@ -131,7 +130,11 @@ def load_event_updates(event_data)
     end
 
     def cleanup
-      run_sync("rm -rf #{remote_command_dir}") if @cleanup_working_dirs
+      if @cleanup_working_dirs
+        assert_remote_command("rm -rf #{remote_command_dir}",
```
[the remark this answers is not in the document]
Comment (adamruzicka) [reply lib/smart_proxy_remote_execution_ssh/runners/polling_script_runner.rb]: Right, now I got what you meant

**Review by ezr-ondrej — APPROVED** [review]
No, I do not and we can add more when we get some feedback on this, sorry I had the foolish idea to understand this completely :shrug:
Lets just go with "it works and looks good from what I see" :+1:

**Comment by adamruzicka** [comment]
> I had the foolish idea to understand this completely 

May I help with this somehow?

**Comment by ezr-ondrej** [comment]
> May I help with this somehow?

Too late, unless it breaks I do not want to understand (as I will not touch it unless I will need to) xD

**Pull request merged**